assets/activation-scripts/activate.d/zdotdir/.zprofile

```diff
@@ -24,11 +25,13 @@ restore_saved_vars() {
     export ZDOTDIR="$_save_ZDOTDIR"
     export _activate_d="$_save_activate_d"
     export _flox_activate_tracer="$_save_flox_activate_tracer"
-    export _flox_env_helper="$_save_flox_env_helper"
     export FLOX_ZSH_INIT_SCRIPT="$_save_FLOX_ZSH_INIT_SCRIPT"
     export _FLOX_ACTIVATION_STATE_DIR="$_save_FLOX_ACTIVATION_STATE_DIR"
     export _FLOX_ACTIVATION_PROFILE_ONLY="$_save_FLOX_ACTIVATION_PROFILE_ONLY"
-    source =("$_flox_env_helper" "zsh")
+    # shellcheck disable=SC1090
+    source <("$_flox_activations" set-env-dirs --shell zsh --flox-env "$FLOX_ENV" --env-dirs "${FLOX_ENV_DIRS:-}")
```

Comment:
note: we can move `set-env-dirs` to the activation script as a followup change, right?

Reply:
Not sure what you mean. Like move it out of Rust?

Reply:
I mean moving `source <("$_flox_activations" set-env-dirs --shell zsh --flox-env "$FLOX_ENV" --env-dirs "${FLOX_ENV_DIRS:-}")` to `assets/activation-scripts/activate` and dropping how we're calling it in so many places

Reply:
Oh, yes. I kept things as close to the previous implementation/behavior as possible to make it easier to review.

Reply:
Cool, it did make it easier to review! Just wanted to make sure that was why